### PR TITLE
✨Add support for compact in Dialog

### LINKED ISSUE #1755: Add compact density for Dialog
- [x] React

- [ ] Design

- [ ] Documentation

### PR DESCRIPTION
**Pull request by laliq**
Subtask of #1755

### CODE REVIEW

File: libraries/core-react/src/components/Dialog/Dialog.tokens.ts [hunk 1]
```diff
@@ -62,4 +63,9 @@ export const dialog: DialogToken = {
       minHeight: '48px',
     },
   },
+  modes: {
+    compact: {
+      height: shape._modes.compact.straight.minHeight,
```
Comment (mimarz) [inline libraries/core-react/src/components/Dialog/Dialog.tokens.ts:68]: Suggest removing this for now as `height` is not used in the component and we also don't know how the "compact" dialog should look like.

**Review by mimarz — CHANGES_REQUESTED** [review]
Looking good.
Looks like thing went abit fast and you missed updating some `styled` declarations in the sub-components to use theme prop instead of global token. 🤪 
As far as I could see,  line 6 in `Actions.tsx`, line 7 & 23 in `CustomContent.tsx` and line 7 in `Title.tsx`

**Review by oddvernes — APPROVED** [review]
LGTM 👍

File: libraries/core-react/src/components/Dialog/Dialog.stories.tsx [hunk 2]
```diff
@@ -180,3 +188,24 @@ export const NoTitle: Story<DialogProps> = () => (
     </Dialog.Actions>
   </Dialog>
 )
+
+export const Compact: Story<DialogProps> = () => {
+  const [density] = useState<Density>('comfortable')
```
Comment (mimarz) [inline libraries/core-react/src/components/Dialog/Dialog.stories.tsx:193]: this should probably be `compact` for testing compact?

**Review by mimarz — APPROVED** [review]
LGTM 👍

**Pull request merged**